### PR TITLE
Add Australia tags and 0% AB test for Ipsos Mori

### PR DESCRIPTION
**Pull request by lucymonie**
## What does this change?
This adds some Ipsos Mori tag mapping for Australia-specific tag pages and fronts so the correct (more specific) tags are passed rather than the generic `ipsosTag=guardian` tag.

In addition, this adds logic that will run the Ipsos Mori script on pages browsed in the AU geo region, but makes this available only via a 0% AB test. When the region is AU, a localised dotmetrics script is loaded.

To force the variant to show, add the following to any dotcom url: `#ab-IpsosMoriAustralia=variant`

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
The Australian team has indicated that this is a very high priority as a lack of tags will lead to lost ad revenue.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by chrislomaxjones — COMMENTED** [review]
Nice! A couple of comments 👇

**Comment by jamesgorrie** [comment]
Part of: https://github.com/guardian/frontend/issues/24980

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts [hunk 1]
```diff
@@ -24,17 +27,25 @@ const loadIpsosScript = () => {
...
-export const init = (): Promise<void> =>
-	getLocale()
+export const init = (): Promise<void> => {
+	const forceIpsosMoriAustraliaTest = isInVariantSynchronous(
+		ipsosMoriAustralia,
+		'variant',
+	);
+
+	return getLocale()
 		.then((locale) => {
```
[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:37]: Thank you so much for this feedback. This code was bothering me and I like the way you've suggested maintaining the existing pattern.

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts [hunk 2]
```diff
@@ -1,16 +1,19 @@
 import { getConsentFor } from '@guardian/consent-management-platform';
 import { getLocale, loadScript, log } from '@guardian/libs';
 import { getInitialConsentState } from 'commercial/initial-consent-state';
+import { isInVariantSynchronous } from 'common/modules/experiments/ab';
+import { ipsosMoriAustralia } from 'common/modules/experiments/tests/ipsos-mori-australia';
 import config from '../../../lib/config';
 import { stub } from './__vendor/ipsos-mori';
 
-const loadIpsosScript = () => {
+const loadIpsosScript = (locale?: string) => {
```
[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:9]: Really good point, and clearer implementation with the first set of changes. Thanks Chris 👍🏻

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:37]: And this sets us up for a smaller change to handle consent for Australia, which we may need to do.

[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:37]: However, it seemed to cause some issue that I'm having a bit of trouble tracking down, so I've kept the awkward chaining as is for the moment. I'd be happy to show the issue that I'm having - it's to do with the consentState that gets passed back from the call to `getInitialConsentState`.

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts [hunk 3]
```diff
@@ -24,22 +26,33 @@ const loadIpsosScript = () => {
...
 			} else {
-				throw Error('Skipping ipsos outside of GB');
+				throw Error('Skipping ipsos process outside GB or AU');
 			}
 		})
 		.then((state) => {
-			if (getConsentFor('ipsos', state)) {
-				void loadIpsosScript();
+			if (state.aus) {
+				void loadIpsosScript('au');
```
Comment (OllysCoding) [inline static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:48]: I'm guessing that in aus we don't need any additional consent to serve this script?

**Review by OllysCoding — APPROVED** [review]
+1 This looks good to me :)

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts (hunk 3 above)
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:48]: Thanks Olly - good question. Yes, we've got the go ahead to serve this script in the 0% AB test without any further consent check for Australia.

**Review by mxdvl — COMMENTED** [review]
Nice.

You might want to turn your method to `async`/`await`!

**Review by arelra — APPROVED** [review]
Look good! Just some minor refactor suggestions  👍

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:36]: I love this suggestion! That would be a lot easier to work with. As it's a reasonably big change, and this is code we need in the codebase very soon, I'm going to put it in a trello card and circle back. Thank you so much @mxdvl

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts [hunk 4]
```diff
@@ -24,22 +26,33 @@ const loadIpsosScript = () => {
...
+			} else if (getConsentFor('ipsos', state)) {
+				void loadIpsosScript('uk');
```
[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:50]: Yes, very good point @arelra - shall do

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts [hunk 5]
```diff
@@ -24,22 +26,33 @@ const loadIpsosScript = () => {
...
+	);
+
+	return getLocale()
 		.then((locale) => {
-			if (locale === 'GB') {
+			if (
+				locale === 'GB' ||
+				(locale === 'AU' && forceIpsosMoriAustraliaTest)
+			) {
 				return getInitialConsentState();
```
[the remark this answers is not in the document]
Comment (lucymonie) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:41]: Great point - I am postponing some of the work suggested here due to the urgency of getting this code merged. I will circle back to this once the pressure is off 😃

[the remark this answers is not in the document]
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:41]: Of course np, congrats on getting this out!

File: static/src/javascripts/projects/commercial/modules/ipsos-mori.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/ipsos-mori.ts:36]: Sometimes, VSCode will offer you to change it automagically. I've used it in a couple places when refactoring.

**Pull request merged**